mappings/net/minecraft/world/World.mapping

```diff
@@ -20,58 +61,254 @@ CLASS net/minecraft/class_18 net/minecraft/world/World
+		ARG 4 maxX
+		ARG 5 maxY
+		ARG 6 maxZ
+	METHOD method_156 canPlace (IIIIZI)Z
+		ARG 1 blockId
+		ARG 2 x
+		ARG 3 y
+		ARG 4 z
+		ARG 5 fallingBlock
 		ARG 6 side
+	METHOD method_157 setBlockEntity (IIILnet/minecraft/class_55;)V
+		ARG 1 x
+		ARG 2 y
+		ARG 3 z
+		ARG 4 blockEntity
+	METHOD method_158 getLightLevel (IIIZ)I
+		ARG 1 x
+		ARG 2 y
+		ARG 3 z
 	METHOD method_159 setTime (J)V
 		ARG 1 time
+	METHOD method_160 raycast (Lnet/minecraft/class_26;Lnet/minecraft/class_26;)Lnet/minecraft/class_27;
+		ARG 1 start
+		ARG 2 end
+	METHOD method_161 raycast (Lnet/minecraft/class_26;Lnet/minecraft/class_26;Z)Lnet/minecraft/class_27;
+		ARG 1 start
+		ARG 2 end
+	METHOD method_162 raycast (Lnet/minecraft/class_26;Lnet/minecraft/class_26;ZZ)Lnet/minecraft/class_27;
+		ARG 1 start
+		ARG 2 end
+	METHOD method_163 getVisibilityRatio (Lnet/minecraft/class_26;Lnet/minecraft/class_25;)F
+		ARG 1 vec
+		ARG 2 box
+	METHOD method_164 getBrightness (Lnet/minecraft/class_56;III)I
+		ARG 1 type
+		ARG 2 x
+		ARG 3 y
+		ARG 4 z
+	METHOD method_165 updateLight (Lnet/minecraft/class_56;IIII)V
+		ARG 1 lightType
+		ARG 2 x
+		ARG 3 y
+		ARG 4 z
+	METHOD method_166 queueLightUpdate (Lnet/minecraft/class_56;IIIIII)V
+		ARG 1 type
+		ARG 2 minX
+		ARG 3 minY
+		ARG 4 minZ
+		ARG 5 maxX
+		ARG 6 maxY
+		ARG 7 maxZ
+	METHOD method_167 queueLightUpdate (Lnet/minecraft/class_56;IIIIIIZ)V
+		ARG 1 type
+		ARG 2 minX
+		ARG 3 minY
+		ARG 4 minZ
+		ARG 5 maxX
+		ARG 6 maxY
+		ARG 7 maxZ
 	METHOD method_168 canSpawnEntity (Lnet/minecraft/class_25;)Z
 		ARG 1 box
-	METHOD method_169 isBoxInMaterial (Lnet/minecraft/class_25;Lnet/minecraft/class_15;)Z
+	METHOD method_169 isMaterialInBox (Lnet/minecraft/class_25;Lnet/minecraft/class_15;)Z
 		ARG 1 boundingBox
 		ARG 2 material
 	METHOD method_170 updateMovementInFluid (Lnet/minecraft/class_25;Lnet/minecraft/class_15;Lnet/minecraft/class_57;)Z
 		COMMENT Also could be for materials, but updated in later versions to be exclusive to fluids
 		ARG 1 entityBoundingBox
 		ARG 2 fluidMaterial
 		ARG 3 entity
-	METHOD method_172 (Lnet/minecraft/class_54;IIII)V
+	METHOD method_171 canInteract (Lnet/minecraft/class_54;III)Z
+		ARG 1 player
 		ARG 2 x
 		ARG 3 y
 		ARG 4 z
+	METHOD method_172 extinguishFire (Lnet/minecraft/class_54;IIII)V
+		ARG 1 player
+		ARG 2 x
+		ARG 3 y
+		ARG 4 z
+		ARG 5 direction
+	METHOD method_173 worldEvent (Lnet/minecraft/class_54;IIIII)V
+		ARG 1 player
+		ARG 2 event
+		ARG 3 x
+		ARG 4 y
+		ARG 5 z
+		ARG 6 data
+	METHOD method_174 countEntities (Ljava/lang/Class;)I
+		ARG 1 entityClass
+	METHOD method_175 collectEntitiesByClass (Ljava/lang/Class;Lnet/minecraft/class_25;)Ljava/util/List;
+		ARG 1 entityClass
+		ARG 2 box
 	METHOD method_176 getOrCreateState (Ljava/lang/Class;Ljava/lang/String;)Lnet/minecraft/class_58;
 		ARG 1 stateClass
 		ARG 2 id
+	METHOD method_177 getPlayer (Ljava/lang/String;)Lnet/minecraft/class_54;
+		ARG 1 name
 	METHOD method_178 addParticle (Ljava/lang/String;DDDDDD)V
```

Comment:
This method does not seem to have all parameters mapped. After some testing, I can conclude that the first 3 doubles are xyz-coordinates. The other 3 would be more tricky to name due to serving 2 different purposes. They are xyz-velocity values by default except for the "reddust" particle. In that special case, they are RGB values (note that they are added on top of the existing red base colour)

Reply:
Here are some suggestions for parameter names.
For the first 3 doubles: x y z
For the last 3 doubles: specialLogic1 specialLogic2 specialLogic3

Reply:
Could be named velocityX, velocityY and velocityZ in acoordance to Yarn as that is what its used for in most casses and add a Javadoc comment that for redstone dust its used for something different

Reply:
Changed to velocity with a Java doc on special behaviors